plugins/aws/ecs/ecsContainerInsightsEnabled.js

```diff
@@ -0,0 +1,119 @@
+    link: 'https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cloudwatch-container-insights.html',
+    apis: ['ECS:listClusters', 'ECS:describeCluster'],
+
+    run: function(cache, settings, callback){
+
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        async.each(
+            regions.ecs,
+            function(region, rcb){
+                var listClusters = helpers.addSource(cache, source, [
+                    'ecs',
+                    'listClusters',
+                    region
+                ]);
+                if (!listClusters) return rcb();
+
+                if (listClusters.err || !listClusters.data) {
+                    helpers.addResult(
+                        results,
+                        3,
+                        'Unable to query for ECS clusters: ' +
+              helpers.addError(listClusters),
+                        region
+                    );
+                    return rcb();
+                }
+
+                if (listClusters.data.length === 0) {
+                    helpers.addResult(results, 0, 'No ECS clusters present', region);
+                    return rcb();
+                }
+     
+                for (var c in listClusters.data) {
+                    var clsuterARN = listClusters.data[c];
+                    var describeCluster = helpers.addSource(cache, source, [
+                        'ecs',
+                        'describeCluster',
+                        region,
+                        clsuterARN
+                    ]);
+                    var arn = clsuterARN;
+            
+                    if (
+                        !describeCluster ||
+            describeCluster.err ||
+            !describeCluster.data
+                    ) {
+                        helpers.addResult(
+                            results,
+                            3,
+                            'Unable to describe ECS cluster: ' +
+                                helpers.addError(describeCluster),
+                            region,
+                            arn
+                        );
+                        continue;
+                    }
+
+                    if (
+                        describeCluster.data.clusters &&
+                        describeCluster.data.clusters.length
+                    ) {   
+            
+                        for ( var index in describeCluster.data.clusters) {
+             
+                            const cluster = describeCluster.data.clusters[index];
+                            if (cluster.settings.length > 0){ 
+                                for (var item of cluster.settings ){ 
+                                    if (item.name === 'containerInsights' ){
+
+                                        if ( item.value === 'enabled'){
+
+                                            helpers.addResult(
```

Comment:
use a flag like inightsEnabled and set it inside loop and add result outside the loop. Just to make sure we don't generate multiple results for same resource